openapi_python_client/parser/properties/union.py

```diff
@@ -67,16 +75,7 @@ def build(
                 return PropertyError(detail=f"Invalid property in union {name}", data=sub_prop_data), schemas
             sub_properties.append(sub_prop)
 
-        def flatten_union_properties(sub_properties: list[PropertyProtocol]) -> list[PropertyProtocol]:
```

Comment:
Broke this out into a separate helper below, since it doesn't rely on any closure variables.